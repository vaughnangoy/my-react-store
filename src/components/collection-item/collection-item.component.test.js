import React from 'react';
import { cleanup, render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import CollectionItem from './collection-item.component';

describe('CollectionItem', () => {
    afterEach(cleanup);
    it('renders', () => {
        const { asFragment } = render(<CollectionItem />);
        expect(asFragment()).toMatchSnapshot();
    });
    it('should have image, name and price info', () => {
        const props = {
            name: 'Trainers',
            price: '£99.99',
            imageUrl: './imageurl',
        };
        const { getByTestId, getByText } = render(
            <CollectionItem {...props} />,
        );
        expect(getByTestId('image')).toBeInTheDocument();
        expect(getByText(/Trainers/)).toBeInTheDocument();
        expect(getByText('£99.99')).toBeInTheDocument();
    });
});
