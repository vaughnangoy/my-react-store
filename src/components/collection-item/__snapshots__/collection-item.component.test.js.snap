// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionItem renders 1`] = `
<DocumentFragment>
  <div
    class="collection-item"
  >
    <div
      class="image"
      data-testid="image"
      style="background-image: url(undefined);"
    />
    <div
      class="collection-footer"
    >
      <span
        class="name"
      />
      <span
        class="price"
      />
    </div>
  </div>
</DocumentFragment>
`;
